fontconfig/src/lib.rs

```diff
@@ -43,20 +46,23 @@ use crate::fontconfig_sys::fontconfig as sys;
 
 use std::ffi::{CStr, CString};
 use std::mem;
-use std::ops::Deref;
 use std::path::PathBuf;
 use std::ptr;
 use std::str::FromStr;
 
-use sys::{FcBool, FcPattern};
-
 pub use sys::constants::*;
+use sys::{FcBool, FcPattern};
 
 #[allow(non_upper_case_globals)]
 const FcTrue: FcBool = 1;
 #[allow(non_upper_case_globals, dead_code)]
 const FcFalse: FcBool = 0;
 
+/// Handle obtained after Fontconfig has been initialised.
+pub struct Fontconfig {
+    _initialised: (),
```

Comment:
Does the struct need to have this field or is it mostly for documentation purposes?

Reply:
It needs to have it to prevent users of the crate from being able to construct an instance of it without going through the `new` function.

Reply:
Ohh right, can one even create an abstract type in Rust? I always forget.

Reply:
I don't think so. @brendanzab can probably confirm.

Reply:
Yeah, this is how you make an abstract type. You can't really do it in the OCaml way, alas (something I'm sure Graydon is sad about).

Reply:
As I mentioned in a message somewhere else:

>  In Standard ML and OCaml you could make it an abstract type, and in Haskell/Elm/Purescript you could avoid exporting the data constructor, but they all amount to the same thing